src/internal/m365/collection/drive/collections_tree_test.go

```diff
@@ -508,3 +472,942 @@ func (suite *CollectionsTreeUnitSuite) TestCollections_GetTree() {
+							rootAnd(
+								driveItem(id(folder), name(folder), parent(0), rootID, isFolder),
+								driveItem(idx(folder, "chld"), namex(folder, "chld"), parent(0), id(folder), isFolder))),
+						DeltaUpdate: pagers.DeltaUpdate{URL: id(delta)},
+					},
+				},
+			},
+			limiter: newPagerLimiter(control.DefaultOptions()),
+			expect: expected{
+				counts: countTD.Expected{
+					count.TotalFoldersProcessed: 7,
+					count.PagesEnumerated:       3,
+				},
+				err:      require.NoError,
+				treeSize: 4,
+				treeContainsFolderIDs: []string{
+					rootID,
+					id(folder),
+					idx(folder, "sib"),
+					idx(folder, "chld"),
+				},
+				treeContainsTombstoneIDs: []string{},
+			},
+		},
+		{
+			name: "tombstone with unpopulated tree",
+			tree: newFolderyMcFolderFace(nil),
+			enumerator: mock.EnumerateItemsDeltaByDrive{
+				DrivePagers: map[string]*mock.DriveItemsDeltaPager{
+					id(drive): {
+						Pages:       pagesOf(rootAnd(delItem(id(folder), parent(0), idx(folder, "parent"), isFolder))),
+						DeltaUpdate: pagers.DeltaUpdate{URL: id(delta)},
+					},
+				},
+			},
+			limiter: newPagerLimiter(control.DefaultOptions()),
+			expect: expected{
+				counts: countTD.Expected{
+					count.TotalFoldersProcessed:       1,
+					count.TotalDeleteFoldersProcessed: 1,
+					count.PagesEnumerated:             1,
+				},
+				err:      require.NoError,
+				treeSize: 2,
+				treeContainsFolderIDs: []string{
+					rootID,
+				},
+				treeContainsTombstoneIDs: []string{
+					id(folder),
+				},
+			},
+		},
+		{
+			name: "tombstone with populated tree",
+			tree: treeWithFolders(),
+			enumerator: mock.EnumerateItemsDeltaByDrive{
+				DrivePagers: map[string]*mock.DriveItemsDeltaPager{
+					id(drive): {
+						Pages:       pagesOf(rootAnd(delItem(id(folder), parent(0), idx(folder, "parent"), isFolder))),
+						DeltaUpdate: pagers.DeltaUpdate{URL: id(delta)},
+					},
+				},
+			},
+			limiter: newPagerLimiter(control.DefaultOptions()),
+			expect: expected{
+				counts: countTD.Expected{
+					count.TotalFoldersProcessed:       1,
+					count.TotalDeleteFoldersProcessed: 1,
+					count.PagesEnumerated:             1,
+				},
+				err:      require.NoError,
+				treeSize: 3,
+				treeContainsFolderIDs: []string{
+					rootID,
+					idx(folder, "parent"),
+				},
+				treeContainsTombstoneIDs: []string{
+					id(folder),
+				},
+			},
+		},
+		{
+			name: "create->delete",
+			tree: newFolderyMcFolderFace(nil),
+			enumerator: mock.EnumerateItemsDeltaByDrive{
+				DrivePagers: map[string]*mock.DriveItemsDeltaPager{
+					id(drive): {
+						Pages: pagesOf(
+							rootAnd(driveItem(id(folder), name(folder), parent(0), rootID, isFolder)),
+							rootAnd(delItem(id(folder), parent(0), rootID, isFolder))),
+						DeltaUpdate: pagers.DeltaUpdate{URL: id(delta)},
+					},
+				},
+			},
+			limiter: newPagerLimiter(control.DefaultOptions()),
+			expect: expected{
+				counts: countTD.Expected{
+					count.TotalFoldersProcessed:       3,
```

Comment:
I've been noodling around where to put that documentation.  There's no clear owner for it, so I don't have one clear spot where I can explain the hidden expectations in a way that's easily discoverable for all maintainers.  If you have any ideas I'm open to them.

Reply:
hmmm maybe some documentation at the definition for the counter keys as to what "Processed" means (similar to what you added in another comment about those keys) with a note that some things may be "processed" multiple times due to how the external service returns data? At least that gives us a starting point for people who look at counter keys

As for the documentation for this particular test, I'm not really sure how to make it easier to grok. I'd like to avoid repeating comments for each test, but a comment on the test itself seems a bit too high-level

Reply:
done